### PR TITLE
SONARJAVA-1976 Display program states and multiple parents

### PR DESCRIPTION
**Pull request by m-g-sonar**

### CODE REVIEW

File: java-frontend/src/main/java/org/sonar/java/se/ExplodedGraph.java [hunk 1]
```diff
@@ -105,12 +105,14 @@ public String toString() {
     public final ProgramState programState;
     @Nullable
     public Node parent;
+    public final List<Node> cacheHits;
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/se/ExplodedGraph.java]: This is poorly named : cache hit has absolutely no significance here... Those are parents nodes.

File: java-frontend/src/test/java/org/sonar/java/viewer/EGViewer.java [hunk 2]
```diff
@@ -42,6 +42,7 @@
 
   private static final ActionParser<Tree> PARSER = JavaParser.createParser(StandardCharsets.UTF_8);
   private final Viewer viewer;
+  private static final boolean SHOW_CACHE = true;
```
Comment (benzonico) [inline java-frontend/src/test/java/org/sonar/java/viewer/EGViewer.java]: Naming! notion of cache should probably not appear anywhere except for the walker : this is implementation detail.

File: java-frontend/src/main/java/org/sonar/java/resolve/JavaSymbol.java [hunk 3]
```diff
@@ -645,6 +645,11 @@ public MethodTree declaration() {
     public boolean isParametrized() {
       return !typeVariableTypes.isEmpty();
     }
+
+    @Override
```
Comment (benzonico) [inline java-frontend/src/main/java/org/sonar/java/resolve/JavaSymbol.java]: missing a test.

File: java-frontend/src/test/java/org/sonar/java/viewer/EGViewer.java (hunk 2 above)
Comment (m-g-sonar) [inline java-frontend/src/test/java/org/sonar/java/viewer/EGViewer.java]: This is an implementation details that I want to see appearing in the EG Viewer, which is a debugging tool. Knowing that a link come from the cache or not seems then important to me.

File: java-frontend/src/main/java/org/sonar/java/resolve/JavaSymbol.java (hunk 3 above)
Comment (m-g-sonar) [inline java-frontend/src/main/java/org/sonar/java/resolve/JavaSymbol.java]: tested

**Pull request merged**